core_impl/src/main/java/io/opencensus/impl/tags/SerializationUtils.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright 2016, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.impl.tags;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.io.ByteStreams;
+import io.opencensus.common.Function;
+import io.opencensus.common.Functions;
+import io.opencensus.internal.VarInt;
+import io.opencensus.tags.Tag;
+import io.opencensus.tags.Tag.TagBoolean;
+import io.opencensus.tags.Tag.TagLong;
+import io.opencensus.tags.Tag.TagString;
+import io.opencensus.tags.TagContext;
+import io.opencensus.tags.TagKey;
+import io.opencensus.tags.TagKey.TagKeyString;
+import io.opencensus.tags.TagValueString;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.BufferUnderflowException;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Iterator;
+
+/**
+ * {@link TagContext} serialization.
+ *
+ * <p>Encoding of tag context information for passing across RPC's:
+ *
+ * <ul>
+ *   <li>Tags are encoded in single byte sequence. The version 0 format is:
+ *   <li>{@code <version_id><encoded_tags>}
+ *   <li>{@code <version_id> == a single byte, value 0}
+ *   <li>{@code <encoded_tags> == (<tag_field_id><tag_encoding>)*}
+ *       <ul>
+ *         <li>{@code <tag_field_id>} == a single byte, value 0 (string tag), 1 (int tag), 2 (true
+ *             bool tag), 3 (false bool tag)
+ *         <li>{@code <tag_encoding>}:
+ *             <ul>
+ *               <li>{@code (tag_field_id == 0) == <tag_key_len><tag_key><tag_val_len><tag_val>}
+ *                   <ul>
+ *                     <li>{@code <tag_key_len>} == varint encoded integer
+ *                     <li>{@code <tag_key>} == tag_key_len bytes comprising tag key name
+ *                     <li>{@code <tag_val_len>} == varint encoded integer
+ *                     <li>{@code <tag_val>} == tag_val_len bytes comprising UTF-8 string
+ *                   </ul>
+ *               <li>{@code (tag_field_id == 1) == <tag_key_len><tag_key><int_tag_val>}
+ *                   <ul>
+ *                     <li>{@code <tag_key_len>} == varint encoded integer
+ *                     <li>{@code <tag_key>} == tag_key_len bytes comprising tag key name
+ *                     <li>{@code <int_tag_value>} == 8 bytes, little-endian integer
```

Comment:
There is no a good or bad answer. Some languages default to little-endian some to big-endian, the majority of the CPUs are little-endian. So we only need to be consistent between all the languages and avoid unnecessary overhead.

@adriancole  Why does it matter for the users? I think it is important only for other developers of the library but not for the users.